src/rust/engine/process_execution/src/local.rs

```diff
@@ -596,6 +601,23 @@ pub trait CapturedWorkdir {
   ) -> Result<BoxStream<'c, Result<ChildOutput, String>>, String>;
 }
 
+/// Updates the Process env.
+///
+/// Mutates the env for the process `req`, replacing any `{chroot}` placeholders with
+/// `workdir_path`.
+///
+/// This matches the behavior of interactive processes executed by the `run` goal.
```

Comment:
See #14386 on this topic (and maybe mention it in a TODO here?)... it would be great to actually apply this logic using the same codepath, but that can wait for a followup.

Reply:
Yeah, I had the same idea, but didn't want to mud the waters for this change (and also, I'm still not at all productive working on the rust code :P )